test/components/link_test.exs

```diff
@@ -0,0 +1,39 @@
+defmodule Surface.Components.LinkTest do
+  use ExUnit.Case
+
+  alias Surface.Components.Link, warn: false
+
+  import ComponentTestHelper
+
+  describe "Without LiveView" do
+    test "creates a link" do
+      code = """
+      <Link text="user" to="/users/1" />
```

Comment:
This looks good as a first version, but have you given thought as to adding support for children instead of `text`?

Reply:
Good catch, I started off with a default slot. I'll give it a whirl.

Reply:
Maybe it's a good idea to rename `text` to `label` and create the default slot the same way as [LiveRedirect](http://surface-demo.msaraiva.io/builtincomponents/LiveRedirect). See source [here](https://github.com/msaraiva/surface/blob/master/lib/surface/components/live_redirect.ex#L48).

Reply:
You mean by always using the `link/2` version that takes a block?

Reply:
I'm not sure if it's intended but this works for children but breaks the other use cases.

```elixir
    children = ~H"""
    <slot />
    """

    ~H"""
    {{ link [to: @to, class: @class] ++ @opts, do: children }}
    """
```

 Is there a way we can know if a slot is being used to match off of? If default slot is used, use `link/2` with do block, else use `link/2` without do block and require `text` property.

Reply:
If the default slot is not set, `@inner_content`, i.e., `assigns[:inner_content]` should be `nil`. Please let me know if it doesn't.

Reply:
This works for all cases so far:

```elixir
  def render(assigns) do
    case assigns.inner_content do
      nil ->
        ~H"""
        {{ link @text, [to: @to, class: @class] ++ @opts }}
        """

      _ ->
        children = ~H"""
        <slot />
        """

        ~H"""
        {{ link [to: @to, class: @class] ++ @opts, do: children }}
        """
    end
  end
```

Reply:
Sweet! 

I wonder, however, why the following doesn't work for both cases:

```elixir
  children = ~H"""
  <slot>{{@text}}</slot>
  """

  ~H"""
  {{ link [to: @to, class: @class] ++ @opts, do: children }}
```
I'll check later. You can go with the current implementation since it works :)

Reply:
Ha well that does work and I like it better so I'm going to steal it! Less is more

Reply:
But it does add superfluous newlines `\n`, any idea why this happens?

Reply:
That is added at the parser level. Every rendered output of the `H` sigil will have a new line, which makes sense most of the time. I created a PR with support for an `i` option to make the newline go away. The idea being that the rendered output is to be inlined.

Not sure if it is a good idea, but it solve this problem.

Reply:
Yea I have a bunch of components that piece together uses of `~H` and I always get a bunch of white space with it. Wasn't sure why theses weren't trimmed.

Reply:
I’d be interested with which cases those might be. I’m not sure my PR makes sense, but maybe it does.

You can try it in your examples, see what you think

Reply:
There are few cases where an extra `\n` may be added. As @zamith pointed out, using `~H"""` is one of them. You should be able to avoid this one by adding `\` to the end of the text as mentioned here https://github.com/msaraiva/surface/pull/72#issuecomment-615835594. Another option is to use `~H"..."` or `~H(...)`, etc.

There are other scenarios where additional spaces might be added too but those cases are related to code injection during translation. Most of them will be tackled in #15.

So, I wouldn't worry much about those extra spaces for now. As soon we have #15 done, we'll certainly come back to this issue.

Reply:
I just realized that adding `\` does not work for any of the `~H`, `~L` and `~E` sigils :(

@mathewdgardner if you want to get rid of those `\n` in the tests, you can replace the code with:

```elixir
  children = ~H"<slot>{{ @label }}</slot>"

  ~H"{{ link [to: @to, class: @class] ++ @opts, do: children }}"
```